### PR TITLE
Add new 'format' function to build MQTT topic from a topic pattern and a set of values

### PR DESCRIPTION
**Pull request by mcfitz2**
This PR adds a new method to the mqtt_regex result that allows the user to build a valid MQTT topic using a topic pattern and a set of values. 

Usage is as follows:
```
var topic = mqtt_regex("/feeder/+feederId/needInit")
var params = {"feederId":"feeder-1"}
topic.format(params)
```

This should return a formatted MQTT topic that looks like `/feeder/feeder-1/needInit`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -36,12 +36,29 @@ function parse(topic) {
...
+		format: make_formatter(topic)
 	};
 	result.exec = exec.bind(result);
 	return result;
 };
 
+function make_formatter(topic) {
+	function formatter(params) {
+		var tokens = tokenize(topic)
+		for (var i = 0; i < tokens.length; i++) {
```
Comment (RangerMauve) [inline index.js:49]: Would you mind refactoring this to use `.filter()` and then `.map()`?

File: index.js [hunk 2]
```diff
@@ -36,12 +36,29 @@ function parse(topic) {
...
+			if (tokens[i][0] == '+' || tokens[i][0] == '#') {
+				let paramName = tokens[i].slice(1)
+				tokens[i] = params[paramName]
+			}
+		}
+		return tokens.join('/')
```
Comment (RangerMauve) [inline index.js:55]: Should there maybe be a `tokens.flat()` call so that `#` parameters can be arrays?

**Review by RangerMauve — CHANGES_REQUESTED** [review]
This looks useful, thank you!

Left a couple of comments around code style and a potential edge case, else it looks good!

File: index.js (hunk 2 above)
Comment (mcfitz2) [reply index.js]: Good point, I wasn't considering the `#` wildcard. Done.

File: index.js (hunk 1 above)
Comment (mcfitz2) [reply index.js]: Refactored to use `.map()` I didn't use filter because I still want to output any tokens that aren't wildcards.

**Review by RangerMauve — APPROVED** [review]
Looks good, thank you!

**Comment by RangerMauve** [comment]
Released in 1.1.0. Also added a thing to expand the array for the `#` params if one exists.

e.g.

```JavaScript
const {format} = mqtt_pattern('foo/#bar')

format({bar: [1,2,3]}) === 'foo/1/2/3'
```

**Pull request merged**